cmd/mdatagen/internal/samplereceiver/internal/metadatatest/generated_telemetrytest.go

```diff
@@ -19,15 +21,21 @@ import (
 )
 
 type Telemetry struct {
+	Spanrecorder *tracetest.SpanRecorder
```

Comment:
Spanrecorder -> SpanRecorder?

 Also, it is a bit inconsistent how for metrics we have helpers and not expose the reader directly but here we expose the recorded. We should be consistent on this.

Reply:
updated in https://github.com/open-telemetry/opentelemetry-collector/pull/12106/commits/365a0680018d0bfba509ad3b7dcd768d06bd123f